### PR TITLE
Disable Statsbeat via environment variable

### PR DESCRIPTION
**Pull request by vishweshbankwar**
Adding option to disable Statsbeat via environment variable `APPLICATIONINSIGHTS_STATSBEAT_DISABLED`

TODO: Add E2E test.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs [hunk 1]
```diff
@@ -53,6 +53,14 @@ private static void InitializeStatsbeat(ConnectionVars connectionVars)
                 // Do not initialize statsbeat for statsbeat.
                 if (connectionVars != null && connectionVars.InstrumentationKey != ConnectionStringParser.GetValues(Statsbeat.Statsbeat_ConnectionString_EU).InstrumentationKey && connectionVars.InstrumentationKey != ConnectionStringParser.GetValues(Statsbeat.Statsbeat_ConnectionString_NonEU).InstrumentationKey)
                 {
+                    var disableStatsbeat = Environment.GetEnvironmentVariable("APPLICATIONINSIGHTS_STATSBEAT_DISABLED");
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:56]:
```suggestion
                    var disableStatsbeat = Environment.GetEnvironmentVariable("APPLICATIONINSIGHTS_STATSBEAT_ENABLED");
```

Just my preference :)

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:56]: I need to propose that in spec first :)

Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:56]: Should we check this logic in constructor, before calling `InitializeStatsbeat`?

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:56]: We will run it atleast twice in that case, one for user's ikey and one for statsbeat itself. Putting it here prevents checking for statsbeat itself.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**